sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2075,368 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+                    return Page.FromValues(result.AsEnumerable(), operation.NextLink, response.GetRawResponse());
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+            async Task<Page<DocumentHealthcareResult>> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(GetHealthcareEntities)}");
+                scope.Start();
+
+                try
+                {
+                    int top = default;
+                    int skip = default;
+
+                    // Extracting Job ID and parameters from the URL.
+                    // TODO - Update with Regex for cleaner implementation
+                    // nextLink - https://cognitiveusw2dev.azure-api.net/text/analytics/v3.1-preview.3/entities/health/jobs/8002878d-2e43-4675-ad20-455fe004641b?$skip=20&$top=0
+
+                    string[] nextLinkSplit = nextLink.Split('/');
+                    // nextLinkSplit = [ 'https:', '', 'cognitiveusw2dev.azure-api.net', 'text', ..., '8002878d-2e43-4675-ad20-455fe004641b?$skip=20&$top=0']
+
+                    string[] jobIdParams = nextLinkSplit.Last().Split('?');
+                    // jobIdParams = ['8002878d-2e43-4675-ad20-455fe004641b', '$skip=20&$top=0']
+
+                    if (jobIdParams.Count() != 2)
+                    {
+                        throw new InvalidOperationException($"Failed to parse element reference: {nextLink}");
+                    }
+
+                    // The Id for the Job i.e. the first index of the list
+                    string jobId = jobIdParams[0];
+                    // '8002878d-2e43-4675-ad20-455fe004641b'
+
+                    // Extracting Top and Skip parameter values
+                    string[] parameters = jobIdParams[1].Split('&');
+                    // '$skip=20&$top=0'
+
+                    foreach (string paramater in parameters)
+                    {
+                        if (paramater.Contains("top"))
+                        {
+                            _ = int.TryParse(paramater.Split('=')[1], out top);
+                            // 0
+                        }
+                        if (paramater.Contains("skip"))
+                        {
+                            _ = int.TryParse(paramater.Split('=')[1], out skip);
+                            // 20
+                        }
+                    }
+
+                    Response<HealthcareJobState> jobState = await _serviceRestClient.HealthStatusAsync(new Guid(jobId), top, skip, operation.ShowStats).ConfigureAwait(false);
+
+                    RecognizeHealthcareEntitiesResultCollection result = Transforms.ConvertToRecognizeHealthcareEntitiesResultCollection(jobState.Value.Results, _idToIndexMap);
```

Comment:
instead of making a global `idToIndex` why not use the one in `HealthcareOperation` class?